File: dart/06-io-operations/io_input.dart
import 'dart:io';

void main() {
  stdout.write('What is your name? ');
  String? name = stdin.readLineSync();

  stdout.write('How old are you? ');
  String? ageInput = stdin.readLineSync();

  // int.parse converts text to a number; fall back to '0' if input was null
  int age = int.parse(ageInput ?? '0');

  // The ?? operator supplies a default when name is null
  print('Hello, ${name ?? 'stranger'}!');
  print('Next year you will be ${age + 1}.');
}

File: nim/06-io-operations/io_input.nim
import std/strutils

stdout.write("What is your name? ")
let name = readLine(stdin)

stdout.write("What is your favorite number? ")
let numText = readLine(stdin)
let num = parseInt(numText)

echo "Hello, ", name, "!"
echo "Your number doubled is ", num * 2
